src/_gettsim/shared.py

```diff
@@ -260,3 +263,42 @@ def remove_group_suffix(col):
         out = out.removesuffix(f"_{g}")
 
     return out
+
+
+Key: TypeVar = TypeVar("Key")
+Out: TypeVar = TypeVar("Out")
+
+
+def join_numpy(
```

Comment:
```suggestion
def join(
```
In case we can be agnostic about whether we use numpy or jax.numpy